e2e/backend_service_update_test.go

```diff
@@ -0,0 +1,200 @@
+	}
+}
+
+func checkBackendService(t *testing.T, ctx context.Context, cloud cloud.Cloud, bsID *cloud.ResourceID, bsPort int) {
+	t.Helper()
+	t.Log("---- Check BackendService ----")
+	gotBS, err := cloud.BackendServices().Get(ctx, bsID.Key)
+	if err != nil {
+		t.Fatalf("cloud.HealthChecks().Get(_, %s) = %v, want nil", bsID.Key, err)
+	}
+
+	if gotBS.Port != int64(bsPort) {
+		t.Errorf("BackendService port mismatch, got: %d want %d", gotBS.Port, bsPort)
+	}
+}
+func TestBackendServiceUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+
+	bsName := "bs-e2e"
+	bsID, err := buildBackendServiceWithLBScheme(graphBuilder, bsName, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildSimpleBackendService(_, %s, _) = %v, want nil", resourceName(bsName), err)
+	}
+
+	serviceName := fmt.Sprintf("https://compute.googleapis.com/v1/projects/%s/global/backendServices/%s", testFlags.project, resourceName(bsName))
+	tcpr1, err := buildTcpRoute(graphBuilder, "route1", serviceName)
+	t.Logf("tcpr1 = %s", pretty.Sprint(tcpr1))
+	if err != nil {
+		t.Fatalf("buildTcpRoute(_, route1, %s) = %v, want nil", serviceName, err)
+	}
+
+	tcpr2, err := buildTcpRoute(graphBuilder, "route2", serviceName)
+	t.Logf("tcpr2 = %s", pretty.Sprint(tcpr1))
+	if err != nil {
+		t.Fatalf("buildTcpRoute(_, route2, %s) = %v, want nil", serviceName, err)
+	}
+
+	graph, err := graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+
+	result, err := plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	ex, err := exec.NewSerialExecutor(result.Actions)
+	if err != nil {
+		t.Logf("exec.NewSerialExecutor(_, _) err: %v", err)
+		return
+	}
+	res, err := ex.Run(ctx, theCloud)
+	if err != nil || res == nil {
+		t.Errorf("ex.Run(_,_) = %v, want nil", err)
+	}
+	t.Cleanup(func() {
+		err := theCloud.TcpRoutes().Delete(ctx, tcpr1.Key)
+		if err != nil {
+			t.Logf("delete TCProute: %v", err)
+		}
+		err = theCloud.TcpRoutes().Delete(ctx, tcpr2.Key)
+		if err != nil {
+			t.Logf("delete TCProute: %v", err)
+		}
+		err = theCloud.BackendServices().Delete(ctx, bsID.Key)
+		if err != nil {
+			t.Logf("delete backend service: %v", err)
+		}
+	})
+	checkTCPRoute(t, ctx, theCloud, tcpr1)
+	checkTCPRoute(t, ctx, theCloud, tcpr2)
+	checkBackendService(t, ctx, theCloud, bsID, 80)
+
+	// update backend service
+	bsID, err = buildBackendServiceWithLBScheme(graphBuilder, bsName, "INTERNAL_MANAGED")
+
+	graph, err = graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+	result, err = plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	expectedActions := []exec.ActionMetadata{
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, bsID)},
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, tcpr1)},
```

Comment:
TCPRoute should be updated not recreated

Reply:
This behaviour is not implemented yet. Please add TODO to change the expectation when it's fixed

Reply:
Done.